Sources/DDGSync/internal/SyncOperation.swift

```diff
@@ -211,10 +212,15 @@ final class SyncOperation: Operation {
         return SyncRequest(feature: dataProvider.feature, previousSyncTimestamp: dataProvider.lastSyncTimestamp, sent: localChanges)
     }
 
-    private func makeHTTPRequest(with syncRequest: SyncRequest, timestamp: Date) throws -> HTTPRequesting {
+    private func makeHTTPRequest(for dataProvider: DataProviding, with syncRequest: SyncRequest, timestamp: Date) throws -> HTTPRequesting {
         let hasLocalChanges = !syncRequest.sent.isEmpty
         if hasLocalChanges {
-            return try requestMaker.makePatchRequest(with: syncRequest, clientTimestamp: timestamp)
+            do {
+                return try requestMaker.makePatchRequest(with: syncRequest, clientTimestamp: timestamp, isCompressed: true)
+            } catch let error as GzipError {
+                dataProvider.handleSyncError(error)
```

Comment:
This will be emitted by `dataProvider.syncErrorPublisher` and handled on the client side (e.g. in `SyncBookmarksAdapter`) by firing a suitable "sync failed" pixel (e.g. `syncBookmarksFailed`). This is acceptable even though sync didn't actually fail here, only the compression failed. We aren't expecting huge volume of compression errors anyway, and if they become an issue we may add a dedicated pixel.